### ISSUE TITLE
bug: the gap  not work

### ISSUE THREAD

**Issue by liuJchun** [naver/egjs-grid#90]
## Description

this is a bug?
desc: the gap now work when loading,only work when loaded completely 
![image](https://user-images.githubusercontent.com/33089174/228498791-d6a2b3ab-8c85-49e3-962e-f8fc3f94389a.png)
![image](https://user-images.githubusercontent.com/33089174/228499940-ca733223-5c44-4352-8e09-d81e439a99fc.png)

## Steps to check or reproduce

use @egjs/react-grid package will show this problem

**Comment by daybrush** [comment]
@liuJchun 

Probably because batching doesn't happen while loading
The following phenomena may occur.

Try to set data-grid-lazy="true" on item element?

**Comment by stepanjakl** [comment]
Hi, I'm encountering an issue with columns in which one of the columns with a percentage width doesn't have enough space to fill out its spot when the gap option is set. 

Here's a [pen](https://codepen.io/stepj/pen/oNazyWd?editors=1000).

**Comment by daybrush** [comment]
@stepanjakl 

align: 'justify', independent of gap.

If so, change it to align: "center", "start", "end".

**Comment by stepanjakl** [comment]
Hi @daybrush, even if I align the columns differently, the layout is not correct. There should be 3 columns instead of 2 as each column has width of `33.333%`. Please see the example in the previous message. Preferably, the script should take column width into consideration when calculating the gap?

**Comment by daybrush** [comment]
@stepanjakl 

33.333%, so the sum of the three is 100%.
Therefore, it is difficult to set the gap to 30 (set 0).

**Comment by stepanjakl** [comment]
Yes, but if I set gap to `0` then there'll be no gap between the columns. I was hoping the script would calculate the gap taking column width into consideration. Does it make sense?

**Comment by stepanjakl** [comment]
I did sort it out this way btw:
```html
<div class="w-full md:w-[calc(50%-(1rem/2))] lg:w-[calc(33.333333%-(1rem/(3/2)))]{% if loop.index === 1 %} lg:pt-40{% elseif loop.index === 3 %} lg:pt-20{% endif %}">
```
But this is not an ideal solution. The grid library should ideally compute the gap without me adjusting the column width.

**Comment by daybrush** [comment]
@stepanjakl 

It can't be helped because you set the width yourself.

If you expect to automatically change the width, try setting it with the following option. Instead, you must input the `column` value directly.
```js
columns: 3,
gap: 30;
align: "stretch",
```

**Comment by stepanjakl** [comment]
Okay, I tried it, but it's overflowing its parent container now. Moreover, is there a way to set column breakpoints for different viewport widths?

<img width="1226" alt="Screenshot 2023-05-09 at 11 53 04" src="https://github.com/naver/egjs-grid/assets/47005900/97ef6388-43b8-42f9-97eb-3f8d70814761">

**Comment by daybrush** [comment]
@stepanjakl 

<img width="1026" alt="image" src="https://github.com/naver/egjs-grid/assets/3433062/82bae58d-e2e4-43dd-9c40-714d238191b1">


Try modifying align to "stretch".

**Comment by stepanjakl** [comment]
Ouh nice, that works, thanks! :)

One more question, how to handle responsivity when it comes number of columns or gap width?

**Comment by stepanjakl** [comment]
Ouh nice, that works, thanks! :)

One more question, how to handle responsivity when it comes number of columns or gap width?

**Comment by daybrush** [comment]
@stepanjakl 

Modify the column value directly through resize event or resizeObserver.

```js
grid.column = 2 ~ 4
```

**Comment by stepanjakl** [comment]
Awesome, thank you. It's great library in any case.
